src/generate_outputs.py

```diff
@@ -1,12 +1,56 @@
 """
-Model adapted from Beniczky S, et. al. A web-based algorithm to rapidly classify seizures for
-the purpose of drug selection. Epilepsia. 2021 Oct;62(10):2474-2484. doi: 10.1111/epi.17039.
-Epub 2021 Aug 22. PMID: 34420206.
-
-Online tool: epipick.org
+Script of functions to generate input matrix.
 """
 
 import numpy as np
+from typing import Mapping, Sequence
+
+FOCAL_EPILEPSY_BILLING_CODES = ['G40.0', 'G40.1', 'G40.2', 'G40.5', 'G41.2']
+GENERALISED_EPILEPSY_BILLING_CODES = [
+    'G40.7', 'G41.1', 'G40.3', 'G40.6', 'G41.0', 'G40.4'
+]
+UNKNOWN_EPILEPSY_BILLING_CODES = ['G40.8', 'G40.9', 'G41.8', 'G41.9']
+EPILEPSY_BILLING_CODES = FOCAL_EPILEPSY_BILLING_CODES + GENERALISED_EPILEPSY_BILLING_CODES + UNKNOWN_EPILEPSY_BILLING_CODES
+
+
+def set_diagnosis(list_of_billing_codes: Sequence[str]):
+    """_summary_
+
+    Args:
+        list_of_billing_codes (Sequence[str]): _description_
+    """
+
+    # Init row for np.array
+    diagnosis_array = np.zeros([1, 5])
+
+    # Populate rows
+    if any([
+            billing_code for billing_code in EPILEPSY_BILLING_CODES
+            if billing_code in list_of_billing_codes
```

Comment:
Optional slight simplification:
```suggestion
            billing_code in EPILEPSY_BILLING_CODES
            for billing_code in list_of_billing_codes
```
Or with `EPILEPSY_BILLING_CODES` and `list_of_billing_codes` swapped, whichever makes more sense. Or even more radically:

```python
if set(list_of_billing_codes).intersection(set(EPILEPSY_BILLING_CODES)):
```

Reply:
So I ran both options on a large dataset recently and found that set() was quite slow compared to the list iterator. Not sure if that was just a fluke and which is actually faster. I thought set() was. It looks cleaner at least. Do you know?

Reply:
I guess it's a case-by-case and would depend on the length of each list

Reply:
Interesting! I wonder why. The theoretical answer from my comp sci studies is that the set lookup should be faster (once you already have your sets at least), because list lookup time complexity is O(n), sets & dicts should always be closer to O(1) because of hashing (Python implements lists as linked lists not arrays from memory).